pkg/vgmanager/vgmanager_controller.go

```diff
@@ -71,18 +75,60 @@ func (r *VGReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Re
 	return res, err
 
 }
+
+var reconcileInterval = time.Minute * 1
+var reconcileAgain ctrl.Result = ctrl.Result{Requeue: true, RequeueAfter: reconcileInterval}
+
+//TODO: Refactor this function to move the ctrl result to a single place
+
 func (r *VGReconciler) reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	lvmCluster := &lvmv1alpha1.LVMCluster{}
-	err := r.Client.Get(ctx, req.NamespacedName, lvmCluster)
+
+	volumeGroup := &lvmv1alpha1.LVMVolumeGroup{}
+	err := r.Client.Get(ctx, req.NamespacedName, volumeGroup)
 	if err != nil {
+		r.Log.Error(err, "failed to get LVMVolumeGroup", "VGName", req.Name)
 		return ctrl.Result{}, err
 	}
 
-	r.Log.Info("listing block devices")
+	//TODO: actually check the node against the nodeSelector.
+	node := &corev1.Node{}
+	nodeMatches, err := NodeSelectorMatchesNodeLabels(node, volumeGroup.Spec.NodeSelector)
+	if err != nil {
+		r.Log.Error(err, "failed to match nodeSelector to node labels", "VGName", volumeGroup.Name)
+		return reconcileAgain, err
+	}
+
+	if !nodeMatches {
+		//Nothing to be done on this node for the VG.
+		r.Log.Info("node does not match selector", "VGName", volumeGroup.Name)
+		return ctrl.Result{}, nil
+	}
+
+	r.Log.Info("listing block devices", "VGName", volumeGroup.Name)
+
 	//  list block devices
 	blockDevices, err := internal.ListBlockDevices(r.executor)
 	if err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to list block devices: %v", err)
+		return reconcileAgain, fmt.Errorf("failed to list block devices: %v", err)
+	}
+
+	// filter out block devices
+	remainingValidDevices, delayedDevices, err := r.filterAvailableDevices(blockDevices)
+	if err != nil {
+		_ = err
+	}
+
+	var matchingDevices []internal.BlockDevice
+	_, matchingDevices, err = filterMatchingDevices(remainingValidDevices, volumeGroup)
+	if err != nil {
+		r.Log.Error(err, "could not filter matching devices", "VGName", volumeGroup.Name)
+		return reconcileAgain, err
+	}
+
+	status := &lvmv1alpha1.VGStatus{
+		Name:   req.Name,
+		Status: lvmv1alpha1.VGStatusReady,
```

Comment:
- IMHO initial vgstatus should be unknown and we try to set it correctly by the end of reconcile 🤔

Reply:
Can this be taken up in a separate PR?